### PR TITLE
Change main file tests to define sources inline and remove tibs testing infrastructure

### PR DESCRIPTION
**Pull request by ahoppen**
Migrate the last tests that used the tibs workspace to define sources inline and remove the testing infrastructure that relied on TIBS.

Resolves https://github.com/apple/sourcekit-lsp/issues/879

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Sources/SKTestSupport/SwiftPMTestWorkspace.swift [hunk 1]
```diff
@@ -78,6 +78,8 @@ public class SwiftPMTestWorkspace: MultiFileTestWorkspace {
       "-Xswiftc", "-index-ignore-system-modules",
       "-Xcc", "-index-ignore-system-symbols",
     ]
-    try await Process.checkNonZeroExit(arguments: arguments)
+    var environment = ProcessEnv.vars
+    environment["SWIFTPM_ENABLE_CLANG_INDEX_STORE"] = "1"
```
Comment (bnbarham) [inline Sources/SKTestSupport/SwiftPMTestWorkspace.swift:82]: Could you add a comment with the issue/radar number for fixing this in SPM?

File: Tests/SourceKitLSPTests/MainFilesProviderTests.swift [hunk 2]
```diff
@@ -19,78 +19,197 @@ import SourceKitLSP
...
-        #include "bridging.h"
-        void d_new(void) { bridging(); }
+  func testMainFileForHeaderInPackageTarget() async throws {
+    let ws = try await SwiftPMTestWorkspace(
+      files: [
+        "MyLibrary/include/MyLibrary.h": """
+        void bridging(void) {
+          int VARIABLE_NAME = 1;
+        }
+        """,
```
Comment (bnbarham) [inline Tests/SourceKitLSPTests/MainFilesProviderTests.swift:29]: Is this how swift-format formats this? I'd personally prefer indented once (or twice)

File: Tests/SourceKitLSPTests/MainFilesProviderTests.swift [hunk 3]
```diff
@@ -19,78 +19,197 @@ import SourceKitLSP
...
+        // swift-tools-version: 5.7
+
+        import PackageDescription
+
+        let package = Package(
+          name: "MyLibrary",
+          targets: [
+            .target(
+              name: "MyLibrary",
+              cSettings: [.define("VARIABLE_NAME", to: "fromMyLibrary"), .unsafeFlags(["-Wunused-variable"])]
```
Comment (bnbarham) [inline Tests/SourceKitLSPTests/MainFilesProviderTests.swift:44]: Took me a little bit to realize why we were doing this - worth a comment IMO (ie. that we're defining `VARIABLE_NAME` here to make sure that the build settings from the main file are being used for the header).

File: Tests/SourceKitLSPTests/MainFilesProviderTests.swift [hunk 4]
```diff
@@ -19,78 +19,197 @@ import SourceKitLSP
...
+        """,
+      build: false
+    )
+
+    _ = try ws.openDocument("shared.h", language: .c)
 
-    mainFilesDelegate.expectation = expectation(description: "main files changed after edit")
-    try ws.buildAndIndex()
+    // Before we build, we shouldn't have an index and thus we don't infer the build setting for 'shared.h' form
+    // 'MyLibrary.c'. Hence, there's we don't have the '-Wunused-variable' build setting and thus no diagnostics.
```
Comment (bnbarham) [inline Tests/SourceKitLSPTests/MainFilesProviderTests.swift:93]:
```suggestion
    // 'MyLibrary.c'. Hence, we don't have the '-Wunused-variable' build setting and thus no diagnostics.
```

File: Tests/SourceKitLSPTests/MainFilesProviderTests.swift [hunk 5]
```diff
@@ -19,78 +19,197 @@ import SourceKitLSP
...
+        "Sources/shared.h": """
+        void bridging(void) {
+          int VARIABLE_NAME = 1;
+        }
+        """,
+        "Sources/MyLibrary/include/MyLibrary.h": "",
+        "Sources/MyLibrary/MyLibrary.c": """
+        #include "$TEST_DIR/Sources/shared.h"
+        """,
+        "Sources/MyFancyLibrary/include/MyFancyLibraryh": "",
```
Comment (bnbarham) [inline Tests/SourceKitLSPTests/MainFilesProviderTests.swift:167]: This file is unused isn't it?

File: Documentation/Development.md [hunk 6]
```diff
@@ -134,81 +134,6 @@ Where "N" configures the log verbosity and is one of the following numbers: 0 (e
 
 As much as is practical, all code should be covered by tests. New tests can be added under the `Tests` directory and should use `XCTest`. The rest of this section will describe the additional tools available in the `SKTestSupport` module to make it easier to write good and efficient tests.
```
Comment (bnbarham) [inline Documentation/Development.md:136]: I assume your opinion here is that the tests are now obvious enough without any description?

**Review by hamishknight — APPROVED** [review]

File: Tests/SourceKitLSPTests/MainFilesProviderTests.swift (hunk 2 above)
Comment (ahoppen) [reply Tests/SourceKitLSPTests/MainFilesProviderTests.swift:29]: Yes, this is formatted using swift-format.

File: Tests/SourceKitLSPTests/MainFilesProviderTests.swift (hunk 3 above)
Comment (ahoppen) [reply Tests/SourceKitLSPTests/MainFilesProviderTests.swift:44]: I mentioned that we use `VARIABLE_NAME` to communicate the build settings in all test cases now.

File: Tests/SourceKitLSPTests/MainFilesProviderTests.swift (hunk 5 above)
Comment (ahoppen) [reply Tests/SourceKitLSPTests/MainFilesProviderTests.swift:167]: SwiftPM refuses to build if a C target does not have an `include` directory. And in the current infrastructure, the only way to create a directory is to put a file in there.

File: Documentation/Development.md (hunk 6 above)
Comment (ahoppen) [reply Documentation/Development.md:136]: My opinion here is that we should write more documentation that’s up-to-date.

This paragraph in particular doesn’t add any value IMO. It’s mostly just: There should be tests and we use XCTest, which, I think is obvious.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Tests/SourceKitLSPTests/MainFilesProviderTests.swift (hunk 5 above)
Comment (bnbarham) [reply Tests/SourceKitLSPTests/MainFilesProviderTests.swift]: I see. That's... okay. Could just add a `dummy` or something though (to make that more clear)

Comment (ahoppen) [reply Tests/SourceKitLSPTests/MainFilesProviderTests.swift]: Good idea. Renamed to `dummy.h`.

File: Documentation/Development.md (hunk 6 above)
Comment (ahoppen) [reply Documentation/Development.md:136]: OK, for those, I would say that 

> As much as is practical, all code should be covered by tests. New tests can be added under the `Tests` directory and should use `XCTest`.

is obvious and not something that needs to be mentioned in documentation.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**